spec/models/page_spec.rb

```diff
@@ -48,6 +51,12 @@
       expect(Tag.first.pages).to match_array []
     end
 
+    it 'reflects assigned tags in the tag_names property' do
+      page = Page.create!(page_params.merge({tag_ids: Tag.last(2).map(&:id)}))
```

Comment:
you want to use FactoryGirl here to create the page - `page = create :page, tags: Tag.last(2)`